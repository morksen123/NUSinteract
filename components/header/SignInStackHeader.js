import { Appbar } from 'react-native-paper';

import { useContext, useState, View } from 'react';

import { UserContext } from '../../contexts/userContext';

import { supabase } from '../../utils/supabase';

import CustomModal from '../Dialog/CustomModal';

import CustomBadge from '../badge/badge';

const SignInStackHeader = (props) => {

  const { navigation, back } = props

  const { setUser } = useContext(UserContext);

  const [showModal, setShowModal] = useState(false)

  const signOutHandler = async () => {
    const { event } = await supabase.auth.signOut()
    setUser(null)
    console.log(event)
  }
  
  const onPressAddHandler = () => {
    navigation.navigate("Home")
    navigation.navigate("Request")
  }

  return (
    
    <Appbar.Header style={{backgroundColor: '#9E89FE'}} dark={false}>
    {back ? <Appbar.BackAction onPress={navigation.goBack} /> : null}
            <Appbar.Content title=''/>
            <Appbar.Action icon="account-multiple-plus" onPress={onPressAddHandler} />
            <CustomBadge/>
            <Appbar.Action icon="logout" onPress={() => setShowModal(true)}/>
            { 
              showModal && 
              <CustomModal
                  onDoneHandler={signOutHandler}
                  onCancelHandler={() => setShowModal(false)}
                  body={'Logout from NUSinteract?'}
                  title={'Confim Logout'}
              /> 
          }
    </Appbar.Header>

  );
};

export default SignInStackHeader;