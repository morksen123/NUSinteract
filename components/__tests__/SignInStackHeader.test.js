import React from "react";
import renderer from 'react-test-renderer';
import SignInStackHeader from '../header/SignInStackHeader';


test('renders correctly', () => {
    const tree = renderer.create(<SignInStackHeader/>).toJSON();
    expect(tree).toMatchSnapshot();
  });