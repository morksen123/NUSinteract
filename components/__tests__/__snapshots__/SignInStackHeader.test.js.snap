// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityRole="progressbar"
  accessibilityState={
    Object {
      "busy": true,
    }
  }
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "height": 24,
        "opacity": 1,
        "width": 24,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "height": 24,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 24,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "height": 12,
              "overflow": "hidden",
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 24,
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "rotate": "-165deg",
                  },
                ],
                "width": 24,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 12,
                  "overflow": "hidden",
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "borderColor": "#6200ee",
                    "borderRadius": 12,
                    "borderWidth": 2.4,
                    "height": 24,
                    "width": 24,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "height": 24,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 24,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "height": 12,
              "overflow": "hidden",
              "top": 12,
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 24,
                "transform": Array [
                  Object {
                    "translateY": -12,
                  },
                  Object {
                    "rotate": "345deg",
                  },
                ],
                "width": 24,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 12,
                  "overflow": "hidden",
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "borderColor": "#6200ee",
                    "borderRadius": 12,
                    "borderWidth": 2.4,
                    "height": 24,
                    "width": 24,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
